### ISSUE TITLE
Use of markdown headings in roxygen with @md is not caught or handled

### ISSUE THREAD

**Issue by tbates** [r-lib/roxygen2#804]
If one uses a markdown heading ( # ) in roxygen with` @md` tag, the heading is generated, but causes a warning (Unknown xml node: heading) on package load.

EXPECTED: Either the currently invalid heading code is caught, or rendered as something legal.

<details>

Here's an example chunk of roxygen and a tmp function which if included in a package will generate the illegal xml warning.

```Rsplus
#' Headings in Rd are not flagged, but don't work
#'
#' @description test function for headings
#' ## Markdown heading will generate "1: Unknown xml node: heading" warning message on load.
#' 
#' Should either flag the heading code as an error, or render something supported by Rd.
#' 
#' *note*: Add fn to a package and load to see the warning
#'
#' @return - 
#' @export
#' @md
tmp <- function(x) {
    just a demo
}

```

version:
```Rsplus
> packageVersion("roxygen2")
[1] ‘6.1.0’
```

</details>

**Comment by hadley** [comment]
Minimal reprex:

``` r
library(roxygen2)
x <- roc_proc_text(rd_roclet(), "
  #' title
  #' @md
  #' @description
  #' ## Heading 2
  f <- function() {}
")[[1]]
#> Warning: Unknown xml node: heading
```

<sup>Created on 2019-08-22 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by gaborcsardi** [comment]
This was never supposed to work, use @section to create sections.

**Comment by tbates** [comment]
maybe
> Markdown headings not supported. Replace with `@section` to create sections.

**Comment by tbates** [comment]
Nice!
Has gone from 4 unallocated notes about xml heading not supported, to a full line-linked check!!

```R
Updating umx documentation
Warning: @param [umx/R/build_run_modify.R#4271]: Use of inline HTML is not currently supported
Warning: @details [umx/R/lavanify2ram.R#15]: Use of markdown headings is not currently supported
Warning: @details [umx/R/lavanify2ram.R#15]: Use of inline HTML is not currently supported
Warning: @details [umx/R/lavanify2ram.R#15]: Use of inline HTML is not currently supported
Warning: @description [umx/R/misc_and_utility.R#90]: Use of inline HTML is not currently supported
Warning: @description [umx/R/misc_and_utility.R#90]: Use of inline HTML is not currently supported
Warning: @param [umx/R/misc_and_utility.R#105]: Use of inline HTML is not currently supported
Warning: @description [umx/R/misc_and_utility.R#1]: Use of markdown headings is not currently supported
Warning: @param [umx/R/misc_and_utility.R#1310]: Use of markdown headings is not currently supported
Warning: @details [umx/R/misc_and_utility.R#1]: Use of inline HTML is not currently supported
Warning: @description [umx/R/misc_and_utility.R#1]: Use of inline HTML is not currently supported
Warning: @details [umx/R/umx_build_umxSexLim.R#1]: Use of markdown headings is not currently supported
```

**Comment by tbates** [comment]
FYI, lack of line numbers (nearly always `#1`) and non-printing of the offending string means it's hard to find.
e.g. these are each 10,000 line files... and the markdown heading is #....
```R
Updating umx documentation
Warning: @description [R/misc_and_utility.R#1]: Use of markdown headings is not currently supported
Warning: @details [R/misc_and_utility.R#1]: Use of inline HTML is not currently supported
Warning: @description [R/misc_and_utility.R#1]: Use of inline HTML is not currently supported
Warning: @details [R/umx_build_umxSexLim.R#1]: Use of markdown headings is not currently supported
```

**Comment by tbates** [comment]
except that 99% of hits are in examples:
> #' # ============================================
#' # = 1. Here's a simple example with raw data =
#' # ============================================
#' # 2. Use parameters to see the parameter estimates and labels

**Comment by gaborcsardi** [comment]
@tbates FYI, you'll be able to use markdown headings, once these are merged: #907, #908.
